20250117-actors-pubsub.md

```diff
@@ -0,0 +1,140 @@
+# Actor PubSub Implementation Proposal
+
+- Author(s): @joshvanl
+
+## Overview
+
+This proposal introduces PubSub capabilities for Dapr Actors, enabling actors to subscribe to messages within their type for a particular topic.
+Additionally, actors of a specific type will have the ability to publish messages to a topic.
+A single designated PubSub component will be marked as the actor PubSub, similar to how an actor state store is defined.
+There can be only one actor PubSub state store per Dapr deployment.
+
+## Background
+
+Currently, Dapr provides PubSub functionality for applications but lacks native PubSub integration for actors.
+This limitation means that actors cannot communicate efficiently via publish-subscribe messaging patterns within their type.
+By enabling Actor PubSub, actors can directly subscribe to and publish messages to specific topics, improving scalability and communication efficiency.
+
+## Implementation Details
+
+### Actor PubSub Registration
+
+A single PubSub component will be designated as the actor PubSub, similar to how an actor state store is configured.
+The metadata configuration will follow the existing pattern:
+
+```yaml
+metadata:
+  - name: actorpubsub
+    value: "true"
+```
+
+Only one PubSub component can be marked as the actor PubSub.
+
+### Actor Subscriptions
```

Comment:
Maybe we don't need a special subscription. If you add the actor type and id in the cloudevent payload, routing can be done automatically from sidecar without the need of a new API.